regression_em/model.py

```diff
@@ -1,18 +1,258 @@
-import scipy.sparse as sp
-from sklearn.linear_model import LinearRegression
+"""Package to perform regression EM algorithm self.
+For the detail of this algortithm,
+see [https://static.googleusercontent.com/media/research.google.com/ja//pubs/archive/46485.pdf].
+"""
+from random import random
+from typing import Sequence, Tuple
 
+import numpy as np
+from sklearn.linear_model import LinearRegression, Ridge
+from sklearn.base import BaseEstimator
 
-class RegressionEm:
-    def fit(self):
-        pass
+# Define type.
+RegParam = Tuple[np.ndarray, float]
 
-    def predict(self):
-        return 1
 
+class RegressionEM(BaseEstimator):
+    """Regression EM can estimate latent factors based on below model.
 
-if __name__ == "__main__":
-    rem = RegressionEm()
+       Outcome = Left latent * Right latent
 
-    x = sp.rand(100, 100)
-    lr = LinearRegression()
-    print(f"Prediction results: {rem.predict()}")
+       For the detail of Regression EM algortithm,
+       see [https://static.googleusercontent.com/media/research.google.com/ja//pubs/archive/46485.pdf].
+       "Position Bias Estimation for Unbiased Learning to Rank in Personal Search"
+       Xuanhui Wang et al.
+
+       :Parameters:
+
+       alpha : float, default=1
+       Weights associated with regularization term
+
+       max_iter : int, default=100
+       Maximum number of iterations taken for the solvers to converge
+
+       epsilon : float, default=1e-10
+       Tiny value for avoiding zero devide
+
+       with_sample_weights : bool, default = False
+       Indicator to use imbalance weight
+
+       :Example:
+
+       >>> import RegressionEM
+       >>> left_feat = array([[0.17843638, 0.09311004, 0.89600447, ...],
+       [0.55349066, 0.83427622, 0.34841103, ...],
+       ...,
+       [0.22199485, 0.19540406, 0.02678277, ...],
+       [0.62612729, 0.71996384, 0.66445362, ...]])
+       >>> right_feat = array([[..., 0.6177678 , 0.69322733, 0.95146727],
+       [..., 0.96681348, 0.79037145, 0.45834361],
+       ...,
+       [..., 0.64773992, 0.86541352, 0.04755084],
+       [..., 0.37910497, 0.44344932, 0.48168189]])
+       >>> X = np.hstack([left_feat, right_feat])
+       >>> X = array([[0.17843638, 0.09311004, 0.89600447, ..., 0.6177678 , 0.69322733, 0.95146727],
+       [0.55349066, 0.83427622, 0.34841103, ..., 0.96681348, 0.79037145, 0.45834361],
+       ...,
+       [0.22199485, 0.19540406, 0.02678277, ..., 0.64773992, 0.86541352, 0.04755084],
+       [0.62612729, 0.71996384, 0.66445362, ..., 0.37910497, 0.44344932, 0.48168189]])
+       >>> y = array([False, False, False, ...,  True, False,  True])
+       >>> rem = RegressionEM(max_iter=100, with_sample_weights=True, alpha=1)
```

Comment:
Add split_index parameter to `__init__` because this value must be immutable while training and prediction.

Reply:
define split_index in `__init__` func.

Reply:
No index parameter in this example.